core/src/test/scala/org/apache/spark/shuffle/sort/BypassMergeSortShuffleWriterSuite.scala

```diff
@@ -169,6 +170,41 @@ class BypassMergeSortShuffleWriterSuite extends SparkFunSuite with BeforeAndAfte
     assert(taskMetrics.memoryBytesSpilled === 0)
   }
 
+  test("only generate temp shuffle file for non-empty partition") {
+    // Using exception to test whether only non-empty partition creates temp shuffle file,
```

Comment:
`temporaryFilesCreated` is only cleared in `afterAll()`, so it should contain every file which was created even after those files are deleted from the disk. Thus, I think we can simplify this test a bit by not relying on the exception here: just check that `temporaryFilesCreated.size === 3` while `temporaryFilesCreated.count(_.exists()) === 0` after a successful stop.